### ISSUE TITLE
asciidoc-coalescer.rb ignores attributes in file

### ISSUE THREAD

**Issue by tkfu** [asciidoctor/asciidoctor-extensions-lab#62]
This is a very minor issue, since it's got such an easy workaround, but I'm reporting it anyway. 

If I have a doc with conditional formatting based on ifdef::[], the script ignores the attributes set in the file. They have to be passed via the command line. I would expect the following file to give me the english version when passed through the script, but it just gives me blank output. If I do `ruby asciidoc-coalescer.rb -a en file.ad`, however, I get the correct output.

```
:en:

ifdef::en[]
== My Document
English version

include::[some-file-en.ad]
endif::en[]

ifdef::pt[]
== Meu Documento
Versão em portugûes

include::[some-file-pt.ad]
endif::pt[]
```

**Comment by mojavelinux** [comment]
The behavior you are observing happens because the preprocessor reader only looks for preprocessor directives. It does not process attribute entries, or the document header.

I just submitted a fix in PR #63. With this change, the script now reads the document header to load all attribute entries defined there. It then passes all the attributes that are defined at the end of the header to the preprocessor reader. Let me know if that works for your use case.

I also documented what the script is doing better.

**Comment by vonas** [comment]
Unfortunately this problem still persists.

I made these changes: https://github.com/vonas/asciidoctor-extensions-lab/commit/b0a42dc7c09fe9412f2bc711a4d5dbb151df5ac7
* [lib/tex-preprocessor.rb](https://github.com/vonas/asciidoctor-extensions-lab/blob/master/lib/tex-preprocessor.rb)
* [lib/tex-preprocessor/extension.rb](https://github.com/vonas/asciidoctor-extensions-lab/blob/master/lib/tex-preprocessor/extension.rb)
* [lib/tex-preprocessor/sample.adoc](https://raw.githubusercontent.com/vonas/asciidoctor-extensions-lab/master/lib/tex-preprocessor/sample.adoc)

I ran the following command with corresponding input and part of the output:

```
asciidoctor -r ./lib/tex-preprocessor.rb lib/tex-preprocessor/sample.adoc
```
```adoc
:first:

ifndef::first[]
Hidden
endif::first[]

:first!:

ifndef::first[]
Visible
endif::first[]
```
```html
<div class="paragraph">
<p>Hidden</p>
</div>
<div class="paragraph">
<p>Visible</p>
</div>
```

The gist of the extension code is this:

```ruby
Extensions.register do
  preprocessor TeXPreprocessor
end

class TeXPreprocessor < Extensions::Preprocessor
  def process document, reader
    return reader if reader.eof?
    reader.unshift_lines reader.read_lines  # this is the culprit
    reader
  end
end
```

Adding `reader.unshift_lines reader.read_lines` breaks the conditional `ifdef` directives.
